neuralop/models/spectral_convolution.py

```diff
@@ -321,9 +336,22 @@ def forward(self, x, indices=0, output_shape = None):
         
         #Compute Fourier coeffcients
         fft_dims = list(range(-self.order, 0))
-        x = torch.fft.rfftn(x.float(), norm=self.fft_norm, dim=fft_dims)
 
-        out_fft = torch.zeros([batchsize, self.out_channels, *fft_size], device=x.device, dtype=torch.cfloat)
+        if self.fno_block_precision == 'half':
+            x = x.half()
+        else:
+            x = x.float()
```

Comment:
Should we remove this one or do you think we need to always explicitly cast here @crwhite14 @rtu715 ?

Reply:
We can remove the "else: x = x.float()"
I just did that in this commit: https://github.com/neuraloperator/neuraloperator/pull/172/commits/bcc52e768c921ee68f3e6620df27367d63928b0d#diff-5ae1e49af12ed16c75135c0043a08575110fd03d4c722a837b60aa0950b31e32L342-L343